File: scripts/sce.c
#include <stdio.h>

int expr1();
int expr2();

int main (int argc, char *args[])
{
    printf("%d\n", (expr1() || expr2()));
    return 0;
}

int expr1()
{
    return 1;  // true
}

int expr2()
{
    printf("Not Short-circuit evaluation.\n");
    return 1;  // true
}

File: scripts/sce.dart
#!/usr/bin/dart

void main() {
    // In Dart, only the boolean value true is true.
    // All other values are treated as false.
    // So, we use the 'and' operator.
    print(expr1() && expr2());
}

String expr1() {
    return 'return value of expr1.';
}

String expr2() {
    print('Not Short-circuit evaluation.');
    return 'return value of expr2.';
}
